Support information for T2B1

grdddj opened the pull request:
Fixes https://github.com/trezor/trezor-firmware/issues/3161, fixes https://github.com/trezor/trezor-firmware/issues/2793:
- change the device names for new internal model names instead of `trezor1` and `trezor2`
- add support information for `TR` - same as `TT`, additionally deleting some coins

matejcik commented on common/defs/README.md:
```diff
@@ -86,7 +86,7 @@ If you want to add a **wallet link**, modify the file [`wallets.json`](wallets.j
 
 We keep track of support status of each built-in coin over our devices. That is
-`trezor1` for Trezor One, `trezor2` for Trezor T, `connect` for [Connect](https://github.com/trezor/connect)
+`t1b1` for Trezor One, `t2t1` for Trezor T, `t2b1` for Trezor R, `connect` for [Connect](https://github.com/trezor/connect)
```
please change to upper-case everywhere where it's externally visible (so dict keys in support.json, argument names, etc. Keeping it lowercase in local variables is ok)

I also think that aesthetically it would be nice if we called it in lowercase originally :) but it's not -- `T2B1` is a fixed four-byte ASCII case-sensitive sequence and we should treat it as such.

grdddj replied:
Renamed all in [b684b89](https://github.com/trezor/trezor-firmware/pull/3240/commits/b684b89d94bd0f3f4fe713e9a7397c1f69550cb7)

matejcik commented on common/tools/support.py:
```diff
@@ -225,17 +227,19 @@ def check(ignore_missing):
-@click.option("--v2", help="Version for TT release (default: guess from latest)")
+@click.option("--t1b1", help="Version for T1 release (default: guess from latest)")
+@click.option("--t2t1", help="Version for TT release (default: guess from latest)")
+@click.option("--t2b1", help="Version for TR release (default: guess from latest)")
```
please convert this to something like `-r T1B1=1.2.0 -r T2T1=2.6.7`, i.e., a multiple-option that specifies a key and a value

grdddj replied:
Done in [16c0911](https://github.com/trezor/trezor-firmware/pull/3240/commits/16c09115560ee792d410394931d0b112ae394f53)

matejcik commented on core/src/apps/ethereum/networks.py.mako:
```diff
@@ -58,7 +58,7 @@ def by_slip44(slip44: int) -> EthereumNetworkInfo:
 # fmt: off
 def _networks_iterator() -> Iterator[NetworkInfoTuple]:
-% for n in sorted(supported_on("trezor2", eth), key=lambda network: (int(network.chain_id), network.name)):
+% for n in sorted(supported_on("t2t1", eth), key=lambda network: (int(network.chain_id), network.name)):
```
this is currently identical across models but we should do the same thing for T2B1 here as we do for coin_info

grdddj replied:
Done in [1512420](https://github.com/trezor/trezor-firmware/pull/3240/commits/1512420557f66045b63527b6b3f0393024a5b889)

matejcik commented on core/src/apps/ethereum/tokens.py.mako:
```diff
@@ -50,7 +50,7 @@ def token_by_chain_address(chain_id: int, address: bytes) -> EthereumTokenInfo |
 
 def _token_iterator(chain_id: int) -> Iterator[tuple[bytes, str, int, str]]:
-% for token_chain_id, tokens in group_tokens(supported_on("trezor2", erc20)).items():
+% for token_chain_id, tokens in group_tokens(supported_on("t2t1", erc20)).items():
```
dtto here

grdddj replied:
Done in [1512420](https://github.com/trezor/trezor-firmware/pull/3240/commits/1512420557f66045b63527b6b3f0393024a5b889)

matejcik commented on core/src/apps/base.py:
```diff
@@ -100,6 +98,15 @@ def get_features() -> Features:
         ]
 
+        # Only some models have special altcoins
+        if utils.INTERNAL_MODEL in ("T2T1",):
```
this should probably be `not utils.MODEL_IS_T2B1`, or create a `MODEL_IS_T2T1` flag

grdddj replied:
Done in [c07d77e](https://github.com/trezor/trezor-firmware/pull/3240/commits/c07d77e831eef3124b259191b237675f9d0db3d0)

grdddj commented:
Rebased on current `master` and addressed the feedback above.

There is one issue, hopefully not urgent, that the `RELEASES_URL = "https://data.trezor.io/firmware/{}/releases.json"` only accepts `1` or `2` - and not e.g. `T2B1`.  Hot-fixing it in [d734041](https://github.com/trezor/trezor-firmware/pull/3240/commits/d73404152f36b88c1513cc0e277ced29a2057a15) - using the same releases for `T2B1` as for `T2T1`. Not sure who can do the changes on `data.trezor.io`'s side? 

A note, I have run the `python support.py release` and it changed the ordering of keys, because of changing the models to `UPPERCASE` - [d4f0267](https://github.com/trezor/trezor-firmware/pull/3240/commits/d4f026793071cca1b3d81c7f2574bbcc023c154d)

matejcik reviewed (commented):
in the help texts and docstrings there is still a lot of (incorrect) lowercase `t2b1` and so on
otherwise LGTM

matejcik commented:
> There is one issue, hopefully not urgent, that the `RELEASES_URL = "https://data.trezor.io/firmware/{}/releases.json"` only accepts `1` or `2` - and not e.g. `T2B1`. Hot-fixing it in [d734041](https://github.com/trezor/trezor-firmware/pull/3240/commits/d73404152f36b88c1513cc0e277ced29a2057a15) - using the same releases for `T2B1` as for `T2T1`. Not sure who can do the changes on `data.trezor.io`'s side?

ah that's because they have `t2t1` in lowercase. as a hotfix instead place `.lower()` on there, i'll try to get it fixed on the data.tio side

grdddj commented:
> in the help texts and docstrings there is still a lot of (incorrect) lowercase `t2b1` and so on

Right, it was also breaking `check-bitcoin-only` script. Fixed in [3f23621](https://github.com/trezor/trezor-firmware/pull/3240/commits/3f23621c49eacac379a0f5d88fb54ce5e3e60ac7)

grdddj commented:
> > There is one issue, hopefully not urgent, that the `RELEASES_URL = "https://data.trezor.io/firmware/{}/releases.json"` only accepts `1` or `2` - and not e.g. `T2B1`. Hot-fixing it in [d734041](https://github.com/trezor/trezor-firmware/pull/3240/commits/d73404152f36b88c1513cc0e277ced29a2057a15) - using the same releases for `T2B1` as for `T2T1`. Not sure who can do the changes on `data.trezor.io`'s side?
> 
> ah that's because they have `t2t1` in lowercase. as a hotfix instead place `.lower()` on there, i'll try to get it fixed on the data.tio side

It does not seem to work for `t2b1` now - https://data.trezor.io/firmware/t2b1/releases.json

matejcik commented:
issue: https://github.com/trezor/data/issues/63

grdddj commented:
Put everything into one commit - with all the previous commit messages left there as a small documentation/explanation

matejcik reviewed (approved):
let's merge as-is, so that we don't block on the URL question

bosomt commented:
**QA OK**

**Info:**
 - Suite version: desktop 23.10.1 (0b3ccfe0b8e9bc2af00609d640bc1be31ab8edaa)
 - Browser: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) TrezorSuite/23.10.1 Chrome/112.0.5615.204 Electron/24.5.1 Safari/537.36
 - OS: MacIntel
 - Screen: 1440x900
 - Device: Trezor T2B1 2.6.2 regular (revision 6e5967af25de87bebd331d697e6ef940e64286ef)
 - Transport: BridgeTransport 2.0.33
